Modules/posixmodule.c

```diff
@@ -4471,9 +4446,32 @@ os__path_splitroot_impl(PyObject *module, path_t *path)
         *p = L'\\';
     }
 
+#if MS_WINDOWS_GAMES
+    /* network share */
+    if (buffer[0] == L'\\' && buffer[1] == L'\\') {
```

Comment:
Check `buffer[0] == L'\\'`. Then if `buffer[1] && buffer[1] != L'\\'`, split the rooted path, e.g. `"\\spam"` -> `('\\', 'spam')`. Otherwise if `buffer[1] == '\\'`, try to split the UNC path.

This also ensures that the code below won't split something like `"\\:\\spam"` incorrectly as a drive path, e.g. as `('\\:\\', 'spam')` instead of `('\\', ':\\spam')`.

Reply:
This should be fixed.